animation_nodes/nodes/fcurve/evaluate_transforms_fcurves.py

```diff
@@ -0,0 +1,41 @@
+import bpy
+from bpy.props import *
+from ... base_types import AnimationNode
+from ... events import executionCodeChanged
+
+frameTypeItems = [
+    ("OFFSET", "Offset", ""),
+    ("ABSOLUTE", "Absolute", "") ]
+
+class EvaluateFCurveNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_EvaluateTransformsFCurvesNode"
+    bl_label = "Evaluate Transforms FCurves"
+
+    frameType: EnumProperty(
+        name = "Frame Type", default = "OFFSET",
+        items = frameTypeItems, update = executionCodeChanged)
+
+    def create(self):
+        self.newInput("FCurve List", "FCurves", "fCurves")
+        self.newInput("Float", "Frame", "frame")
+        self.newOutput("Matrix", "Matrix", "matrix")
+        self.newOutput("Float", "Frame", "outFrame")
+
+    def draw(self, layout):
+        layout.prop(self, "frameType", text = "Frame")
+
+    def getExecutionCode(self, required):
+        yield "evaluationFrame = frame"
+        if self.frameType == "OFFSET":
+            yield "evaluationFrame += self.nodeTree.scene.frame_current_final"
+        yield "t = {'location': [0,0,0], 'rotation_euler': [0,0,0], 'scale': [1,1,1]}"
+        yield "outFrame = 0"
+        yield "for fcurve in fCurves:"
+        yield "    if fcurve.data_path in t:"
+        yield "        t[fcurve.data_path][fcurve.array_index] = fcurve.evaluate(evaluationFrame)"
+        yield "        outFrame = max(outFrame, fcurve.range()[1])"
+        yield "translation = Vector(tuple(t['location']))"
+        yield "rotation = Euler(tuple(t['rotation_euler']))"
+        yield "scale = Vector(tuple(t['scale']))"
```

Comment:
`AN.utils.math.composeMatrix` doesn't really expect vectors, so you can just pass the lists for the scale and location directly, leaving the Euler as is.